### PR TITLE
Emacs native compilation on master

### PR DESCRIPTION
**Pull request by vmiheer**

### CODE REVIEW

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 1]
```diff
@@ -49,10 +50,11 @@ class Emacs(AutotoolsPackage, GNUMirrorPackage):
...
     depends_on('gnutls', when='+tls')
     depends_on('jpeg')
-    depends_on('m4', when="@master:")
-    depends_on('autoconf', when="@master:")
-    depends_on('automake', when="@master:")
-    depends_on('libtool', when="@master:")
+    depends_on('m4', type=('build'), when="@master:")
+    depends_on('autoconf', type=('build'), when="@master:")
+    depends_on('automake', type=('build'), when="@master:")
+    depends_on('libtool', type=('build'), when="@master:")
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/emacs/package.py]: You can just specify `type='build'` (that is, the parentheses aren't necessary).

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 2]
```diff
@@ -49,10 +50,11 @@ class Emacs(AutotoolsPackage, GNUMirrorPackage):
...
+    depends_on('gcc@11.1.0 +strip languages=jit', when="+native")
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/emacs/package.py]: This probably only needs GCC 11 or newer, right? Then `gcc@11:` would be the way to go.

File: var/spack/repos/builtin/packages/emacs/package.py [hunk 3]
```diff
@@ -80,6 +82,9 @@ def configure_args(self):
         if sys.platform == 'darwin':
             args.append('--without-ns')
 
+        if '+native' in spec:
+            args.append('--with-native-compilation')
```
Comment (michaelkuhn) [inline var/spack/repos/builtin/packages/emacs/package.py:87]: When I try to set +native for emacs@27.1, I get:
```
configure: WARNING: unrecognized options: --with-native-compilation
```
Is this limited to some newer versions?

Comment (vmiheer) [reply var/spack/repos/builtin/packages/emacs/package.py:87]: Yes it's only restricted to master branch. And should come in next release.

Comment (michaelkuhn) [reply var/spack/repos/builtin/packages/emacs/package.py:87]: Ah, now I also see it in the title, sorry. :smile: Maybe it should be restricted to `@master`, then?

**Review by michaelkuhn — APPROVED** [review]

**Pull request merged**